synapse/handlers/federation_event.py

```diff
@@ -1291,17 +1296,10 @@ async def _check_event_auth(
         except AuthError as e:
             logger.warning("Failed auth resolution for %r because %s", event, e)
             context.rejected = RejectedReason.AUTH_ERROR
+            return context
```

Comment:
nowhere else in this method sets `context.rejected` (and it can't be set before entry), so this is equivalent to the removed `if not context.rejected` line.